lib/job.js

```diff
@@ -35,10 +35,10 @@ Job.create = function(queue, jobId, data, opts){
 
 Job.fromId = function(queue, jobId){
   return queue.client.hgetallAsync(queue.toKey(jobId)).then(function(jobData){
-    if(jobData){
-      return Job.fromData(queue, +jobId, jobData);
+    if(_.isEmpty(jobData)){
```

Comment:
Remove this. It's just hiding an error (https://github.com/OptimalBits/bull/pull/275/files) and leaves corrupted data in Redis, causing the job processor to get stuck in a loop.